src/Raven.Server/Smuggler/Documents/StreamDestination.cs

```diff
@@ -632,6 +643,52 @@ public Stream GetTempStream()
             }
         }
 
+        private class StreamTimeSeriesActions : StreamActionsBase, ITimeSeriesActions
+        {
+            private readonly DocumentsOperationContext _context;
+
+            public StreamTimeSeriesActions(BlittableJsonTextWriter writer, DocumentsOperationContext context, string propertyName) : base(writer, propertyName)
+            {
+                _context = context;
+            }
+
+            public unsafe void WriteTimeSeries(TimeSeriesItem item)
+            {
+                if (First == false)
+                    Writer.WriteComma();
+                First = false;
+
+                Writer.WriteStartObject();
+
+                Writer.WritePropertyName(nameof(TimeSeriesItem.DocId));
+                Writer.WriteString(item.DocId);
+                Writer.WriteComma();
+
+                Writer.WritePropertyName(nameof(TimeSeriesItem.Name));
+                Writer.WriteString(item.Name);
+                Writer.WriteComma();
+
+                Writer.WritePropertyName(nameof(TimeSeriesItem.ChangeVector));
+                Writer.WriteString(item.ChangeVector, skipEscaping: true);
+                Writer.WriteComma();
+
+                Writer.WritePropertyName(nameof(TimeSeriesItem.Collection));
+                Writer.WriteString(item.Collection);
+                Writer.WriteComma();
+
+                Writer.WritePropertyName(nameof(TimeSeriesItem.SegmentSize));
+                Writer.WriteInteger(item.Segment.NumberOfBytes);
+                Writer.WriteComma();
+
+                Writer.WritePropertyName(nameof(TimeSeriesItem.Baseline));
+                Writer.WriteDateTime(item.Baseline, true);
+
+                Writer.WriteEndObject();
+
+                Writer.WriteMemoryChunk(item.Segment.Ptr, item.Segment.NumberOfBytes);
```

Comment:
Will this yield a proper JSON?

Reply:
Very unlikely, I would say. It just throw the raw bytes there.
I think we need to handle that like attachments, no?